### PR TITLE
Integration/messaging changes

### PR DESCRIPTION
**Pull request by imvinay84**
Below issues are fixed :
- Search by name or address fix on chat history and new chat.
- Fetch public key of searched address. If public key not found, block the link and show inactive.

image

- Can’t send message to yourself

### CODE REVIEW

File: packages/extension/src/components/chatMessage/index.tsx [hunk 1]
```diff
@@ -49,9 +49,9 @@ export const ChatMessage = ({
 
   useEffect(() => {
     setDecryptingState("in-progress");
```
Comment (jas2212) [inline packages/extension/src/components/chatMessage/index.tsx:51]: setDecryptingState --> what is the use of this ?

File: packages/extension/src/graphQL/messages-api.ts [hunk 2]
```diff
@@ -96,6 +97,8 @@ export const deliverMessages = async (
   senderAddress: string,
   targetAddress: string
 ) => {
+  console.log("calling deliever messages","senderAddress",senderAddress,"targetAddress",targetAddress);
```
Comment (jas2212) [inline packages/extension/src/graphQL/messages-api.ts:100]: Please remove unnecessary logs

File: packages/extension/src/graphQL/messages-api.ts [hunk 3]
```diff
@@ -105,22 +108,23 @@ export const deliverMessages = async (
...
-          ],
-        },
-        context: {
-          headers: {
-            Authorization: `Bearer ${state.user.accessToken}`,
           },
-        },
-      });
+        });
+        console.log("calling deliever inside loop",data);
```
Comment (jas2212) [inline packages/extension/src/graphQL/messages-api.ts:127]: please remove logs

File: packages/extension/src/pages/chat/index.tsx [hunk 4]
```diff
@@ -61,6 +61,11 @@ const ChatView = () => {
 
   const [userChats, setUserChats] = useState<MessageMap>({});
   const [inputVal, setInputVal] = useState("");
+  const [open, setIsOpen] = useState(true);
+  const [messagingPublicKey, setMessagingPublicKey] = useState("");
+  const [openDialog, setIsOpendialog] = useState(false);
```
Comment (jas2212) [inline packages/extension/src/pages/chat/index.tsx:66]: openDialog && open why do we need two diffrent states, cant we just achieve it with openDialog

File: packages/extension/src/pages/chat/index.tsx [hunk 5]
```diff
@@ -163,9 +178,9 @@ const ChatView = () => {
       rightRenderer={<SwitchUser />}
     >
       <div className={style.chatContainer}>
-        {/* {!user.accessToken && (
+        {open && openDialog && messagingPublicKey.length == 0 && (
```
Comment (jas2212) [inline packages/extension/src/pages/chat/index.tsx:181]: if we are dependent on this messagingPublicKey.length then we should not need open and opendialog

File: packages/extension/src/pages/chat/users.tsx [hunk 6]
```diff
@@ -18,7 +18,9 @@ const User: React.FC<{
     history.push(`/chat/${contact}`);
   };
```
Comment (jas2212) [inline packages/extension/src/pages/chat/users.tsx:19]: No changes in this file except spacings , please clean this

File: packages/extension/src/pages/newchat/new-chat.tsx [hunk 7]
```diff
@@ -91,36 +106,71 @@ export const NewChat: FunctionComponent = observer(() => {
...
+  );
 
   useEffect(() => {
-    const userAddresses: NameAddress[] = addressBookConfig.addressBookDatas.map(
-      (data) => {
-        return { name: data.name, address: data.address };
-      }
-    );
-    setAddresses(userAddresses);
+    setAddresses(useraddresses);
```
Comment (jas2212) [inline packages/extension/src/pages/newchat/new-chat.tsx:116]: cant this happen directly in usestate , why is useeffect needed

**Review by jas2212 — CHANGES_REQUESTED** [review]

File: packages/extension/src/pages/chat/index.tsx (hunk 4 above)
Comment (UrAvgDeveloper) [reply packages/extension/src/pages/chat/index.tsx:66]: I am fixing this.

File: packages/extension/src/pages/chat/index.tsx (hunk 5 above)
Comment (UrAvgDeveloper) [reply packages/extension/src/pages/chat/index.tsx:181]: I will fix this.

**Review by jas2212 — APPROVED** [review]

**Pull request merged**